desktop/src/views/Pro/SelectPreset/SelectPreset.tsx

```diff
@@ -0,0 +1,196 @@
+        return true
+      }
+
+      if ((preset.spec?.displayName ?? preset.metadata?.name ?? "").includes(searchString)) {
+        return true
+      }
+
+      if (preset.spec?.source.image && preset.spec.source.image.includes(searchString)) {
+        return true
+      }
+
+      return preset.spec?.source.git && preset.spec.source.git.includes(searchString)
+    })
+  }, [templates?.presets, searchString])
+
+  const navigate = useNavigate()
+
+  const createPlain = useCallback(() => {
+    navigate(Routes.toProWorkspaceCreate(host))
+  }, [navigate, host])
+
+  const changeSearchString = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+    setSearchString(e.target.value ? e.target.value : undefined)
+  }, [])
+
+  return (
+    <Box display={"flex"} flexFlow={"column"} w={"full"} mb="20">
+      <BackToWorkspaces />
+      <HStack w={"full"} align={"center"} justifyContent={"space-between"} mt={"4"} mb={"8"}>
+        <VStack w={"full"} align={"start"} justify={"start"}>
+          <Heading fontWeight={"thin"}>Create Workspace</Heading>
+          <Text>
+            Select a preset below or <Link onClick={createPlain}>create a custom workspace</Link>
+          </Text>
+        </VStack>
+        <InputGroup>
+          <InputLeftElement cursor={"text"} onClick={() => searchInputRef.current?.focus()}>
+            <SearchIcon />
+          </InputLeftElement>
+          <Input
+            ref={searchInputRef}
+            value={searchString ?? ""}
+            placeholder={"Filter by name, repo or image"}
+            spellCheck={false}
+            onChange={changeSearchString}
+            bg={"white"}
+          />
+          {searchString && (
+            <InputRightElement>
+              <IconButton
+                onClick={() => setSearchString(undefined)}
+                aria-label={"Clear search"}
+                variant={"ghost"}
+                icon={<AiOutlineCloseCircle />}
+              />
+            </InputRightElement>
+          )}
+        </InputGroup>
+      </HStack>
+      {isTemplatesLoading ? (
+        <Spinner />
+      ) : (
+        <Grid
+          gridTemplateColumns={`repeat(auto-fit, ${gridChildWidth})`}
+          rowGap={"5"}
+          columnGap={"5"}
+          w={"full"}>
+          {!searchString && (
+            <Box
+              onClick={createPlain}
+              display={"flex"}
+              cursor={"pointer"}
+              flexDir={"column"}
+              alignItems={"center"}
+              justifyContent={"center"}
+              h={gridChildHeight}
+              border={"1px"}
+              borderColor={"divider.main"}
+              boxSizing={"border-box"}
+              borderRadius={"4px"}>
+              <Button variant={"outline"} colorScheme={"primary"}>
+                New Custom Workspace
+              </Button>
+            </Box>
+          )}
+          {filteredPresets.map((preset, index) => (
+            <PresetBox
+              key={index}
```

Comment:
super nit: `preset.metadata.name!` (it's guaranteed to be set by k8s)

Reply:
Done